src/hmac.rs

```diff
@@ -79,3 +82,25 @@ where
     let keypair = K::from(secret_key);
     Ok(keypair)
 }
+
+////////////////////////////////////////////////////////////////////////
+/// HMAC-SHA256 based functions
+
+pub fn hmac(key: &[u8], message: &[u8]) -> Digest {
+    let mut hash = Hmac::<sha2::Sha256>::new_from_slice(key).unwrap();
+    hash.update(message);
+    let output: [u8; DIGEST_LEN] = hash.finalize().into_bytes().as_slice().try_into().unwrap();
+    Digest::new(output)
+}
+
+pub fn hkdf(
+    ikm: &[u8],
+    salt: &[u8],
+    info: &[u8],
+    output: &mut [u8],
```

Comment:
I think this is not aligned with `hmac`'s output. Generally I'd avoid providing output as input. It's very common that end-user won't know with what [u8] size to initiate it, and it's generally a a good hygiene to avoid mutable arrays for crypto outputs. TL;DR let's do the same with `hmac` where HKDF output array is initiated internally, then returned.